xla/pjrt/gpu/se_gpu_pjrt_client_test.cc

```diff
@@ -1779,6 +1779,81 @@ TEST(StreamExecutorGpuClientTest, ExecutablePinnedHostOutputMemoryKindTest) {
+      ag1 = f32[4096,32,128]{2,1,0} all-gather(p1), channel_id=2, replica_groups=[1,8]<=[8], dimensions={0}, use_global_device_ids=true
+      ag2 = f32[4096,8,128]{2,1,0} all-gather(p2), channel_id=3, replica_groups=[1,8]<=[8], dimensions={0}, use_global_device_ids=true
+      ag3 = f32[4096,14336]{1,0} all-gather(p3), channel_id=4, replica_groups=[1,8]<=[8], dimensions={0}, use_global_device_ids=true
+      
+      ar0 = f32[1024]{0} all-reduce(p4), channel_id=5, to_apply=region_0
+      ar1 = f32[1]{0} all-reduce(p5), channel_id=6, to_apply=region_0
+      
+      // Regular operations with default memory space
+      add0 = f32[512,128]{1,0} add(p0, p0)
+      add1 = f32[512,32,128]{2,1,0} add(p1, p1)
+      add2 = f32[512,8,128]{2,1,0} add(p2, p2)
+      
+      // Mix of all-gather results (memory space 1) and regular tensors (memory space 0)
+      ROOT tuple = (f32[4096,128]{1,0}, f32[4096,32,128]{2,1,0}, f32[4096,8,128]{2,1,0}, f32[4096,14336]{1,0},
+                    f32[1024]{0}, f32[1]{0}, f32[1024]{0}, f32[1]{0},
+                    f32[512,128]{1,0}, f32[512,32,128]{2,1,0}, f32[512,8,128]{2,1,0}, f32[512,14336]{1,0},
+                    f32[4096,128]{1,0}, f32[4096,32,128]{2,1,0}, f32[4096,8,128]{2,1,0}, f32[4096,14336]{1,0},
+                    f32[1024]{0}, f32[1]{0}, f32[1024]{0}, f32[1]{0},
+                    f32[512,128]{1,0}, f32[512,32,128]{2,1,0}, f32[512,8,128]{2,1,0}, f32[512,14336]{1,0},
+                    f32[4096,128]{1,0}, f32[4096,32,128]{2,1,0}, f32[4096,8,128]{2,1,0}, f32[4096,14336]{1,0},
+                    f32[1024]{0}, f32[1]{0}, f32[1024]{0}, f32[1]{0},
+                    f32[512,128]{1,0}, f32[512,32,128]{2,1,0}, f32[512,8,128]{2,1,0}, f32[512,14336]{1,0},
+                    f32[4096,128]{1,0}, f32[4096,32,128]{2,1,0}, f32[4096,8,128]{2,1,0}, f32[4096,14336]{1,0},
+                    f32[1024]{0}, f32[1]{0}, f32[1024]{0}, f32[1]{0},
+                    f32[512,128]{1,0}, f32[512,32,128]{2,1,0}, f32[512,8,128]{2,1,0}, f32[512,14336]{1,0},
+                    f32[4096,128]{1,0}, f32[4096,32,128]{2,1,0}, f32[4096,8,128]{2,1,0}, f32[4096,14336]{1,0})
+                   tuple(ag0, ag1, ag2, ag3, ar0, ar1, ar0, ar1,
+                         p0, p1, p2, p3, ag0, ag1, ag2, ag3,
+                         ar0, ar1, ar0, ar1, add0, add1, add2, p3,
+                         ag0, ag1, ag2, ag3, ar0, ar1, ar0, ar1,
+                         p0, p1, p2, p3, ag0, ag1, ag2, ag3,
+                         ar0, ar1, ar0, ar1, add0, add1, add2, p3,
+                         ag0, ag1, ag2, ag3)
+    }
+  )";
+
+  TF_ASSERT_OK_AND_ASSIGN(
+      auto executable,
+      CompileExecutable(kProgramWithCollectiveAndTuple, *client, options));
+
+  TF_ASSERT_OK_AND_ASSIGN(
+      auto memory_stats, executable->GetExecutable()->GetCompiledMemoryStats());
+  EXPECT_GE(memory_stats.output_size_in_bytes, 0);
+  EXPECT_GE(memory_stats.host_output_size_in_bytes, 0);
+  EXPECT_GE(memory_stats.peak_memory_in_bytes, 0);
+}
+
```

Comment:
Can you also add a test case with many tuples. The memory statistics should show the memory usage of tuples, which I think it does not with your current change.